v0.96.8

Wyqer opened the pull request:
| Q | A |
| --- | --- |
| Needs wipe? | yes |

### Description:
Pull Request for the next update.
Currently in draft state as a "not binding ToDo list".
We might add or remove some points in the following list during the work on that update.

### Content:
- [x] (Re)organize the structure of the files/folders (scripts folder and reorganize functions from functions folder)
- ~~[ ] Equalize names of global variables in the whole framework~~partially done
- [x] Add proper headers to presets
- [ ] Add AA/Mortar spawning to enemy bases as constant units (maybe with dynamic simulation) to create a more immersive threat in enemy territory. And to have a strategic approach to take towers and bases before attacking a city (which would cause mortar support from bases and angry civs for destroyed buildings)
- [x] Add a carry function as replacement for the crate pushing (enabled, if no ace is loaded)
- [ ] Add a similar function to escort prisoners. [Example from 0.97.0](https://github.com/KillahPotatoes/KP-Liberation/blob/Rewrite-icebox/Missionframework/modules/28_captive/fnc/fn_captive_escort.sqf)
- [ ] Add KP cratefiller for the "Salvage Depot" and rename it to a logistic station or similar
- [ ] Improve JIP `remoteExec` by not using objects as jipID as it's not stackable
- [ ] Ensure reinitialization of Zeus modules and/or create zeus completely via scripts like in 0.97.0.
- [ ] Remove trigger zones in SQMs, play music via script and open respawn dialog via EH.

Wyqer committed:
WIP state of config organizing

Wyqer committed:
Organized config file

Wyqer committed:
`GRLIB_` prefix to `KPLIB_`

Wyqer committed:
`KP_liberation_` prefix to `KPLIB_`

Wyqer committed:
Update Changelog

Wyqer committed:
Adjust description.ext maxPlayers to count HCs
closes #784

Wyqer committed:
Reorganized arsenal presets and preset variables

Wyqer committed:
Merge pull request #783 from KillahPotatoes/0.96.8-Files_and_variables_refactor

0.96.8 Config file organized and unified prefix

Wyqer committed:
Removed not used deathscreen hpp and paa

Wyqer committed:
Equalize variable names in opfor presets

Wyqer committed:
Headers and variable name changes in enemy presets

Wyqer committed:
changed preset file names to lowercase

Wyqer committed:
Delete CUP_AFRF_MSV.sqf

Wyqer committed:
Delete CUP_BAF_Desert.sqf

Wyqer committed:
Delete CUP_BAF_Woodland.sqf

Wyqer committed:
Delete CUP_CDF.sqf

Wyqer committed:
Delete CUP_ChDKZ.sqf

Wyqer committed:
Delete CUP_SLA.sqf

Wyqer committed:
Delete SLA.sqf

Wyqer committed:
Delete CUP_ACR_Desert.sqf

Wyqer committed:
Delete CUP_ACR_Woodland.sqf

Wyqer committed:
Delete CUP_BAF_Desert.sqf

Wyqer committed:
Delete CUP_BAF_Woodland.sqf

Wyqer committed:
Delete CUP_CDF.sqf

Wyqer committed:
Delete CUP_ChDKZ.sqf

Wyqer committed:
Delete CUP_SLA.sqf

Wyqer committed:
Delete CUP_USA_Desert.sqf

Wyqer committed:
Delete CUP_USA_Woodland.sqf

Wyqer committed:
Delete CUP_USMC_Desert.sqf

Wyqer committed:
Delete CUP_USMC_Woodland.sqf

Wyqer committed:
Delete CUP_NAPA.sqf

Wyqer committed:
Delete RACS.sqf

Wyqer committed:
Fixed not updating opforCount in FSM
closes #788

Wyqer committed:
Update Changelog

Wyqer committed:
Adjustments after variable name changes

Wyqer committed:
Renamed sector arrays

Wyqer committed:
Fix FOB truck get FOB box mass set

Wyqer committed:
Equalize variable name of fob array

Wyqer committed:
Headers for presets

Wyqer committed:
Moved FOB hunting templates

MJVEVERUSKA committed:
Update stringtable.xml

veteran29 committed:
Add changelog entry

veteran29 committed:
Merge pull request #793 from MJVEVERUSKA/v0.96.8

Aktualizovat soubor stringtable.xml

Wyqer committed:
Added `param` prefix to param variables

Wyqer committed:
Changelog update

Wyqer committed:
Update Changelog

Wyqer committed:
Missionstart log output preset strings updated

Wyqer committed:
`save_is_loaded` to `KPLIB_saveLoaded`

Wyqer committed:
`combat_readiness` to `KPLIB_enemyReadiness`

Wyqer committed:
stat variable name for opfor vehicles reset

Wyqer committed:
Adjusted indentation in save_manager

veteran29 committed:
Lower the chance for unarmed BRDM in GM East enemy preset

MJVEVERUSKA committed:
Update stringtable.xml

Wyqer committed:
Merge pull request #796 from MJVEVERUSKA/v0.96.8

Aktualizovat soubor stringtable.xml

Wyqer committed:
Merge pull request #795 from KillahPotatoes/v0.96.8-gm-east

Lower the chance for unarmed BRDM in GM East enemy preset

Wyqer committed:
Merge branch 'v0.96.8' into 0.96.8-param_prefixes

Wyqer committed:
Civilian preset variable rename

Wyqer committed:
Player preset variable rename

Wyqer committed:
Resistance preset variable rename

Wyqer committed:
Code adjustments due to preset variable rename

Wyqer committed:
Merge pull request #798 from KillahPotatoes/0.96.8-preset_variables_rename

[0.96.8] Preset variables rename

Wyqer committed:
Merge branch 'v0.96.8' into 0.96.8-param_prefixes

Wyqer committed:
Merge pull request #794 from KillahPotatoes/0.96.8-param_prefixes

[0.96.8] Prefix for parameter variables

Wyqer committed:
Removed RHS T-14 in presets

Wyqer committed:
Missing `};` in civinfo_escort.sqf
closes #807

Wyqer committed:
Missing `}` in civinfo_task.sqf

Dubjunk committed:
first version of the crate carrying

Dubjunk committed:
crate carrying v2

Dubjunk committed:
updatet crate postition while carrying

Dubjunk committed:
changelog update

Dubjunk committed:
Merge branch 'v0.96.8-carry_ressouce_crates' of https://github.com/KillahPotatoes/KP-Liberation into v0.96.8-carry_ressouce_crates

Wyqer committed:
Merge branch 'v0.96.71' into v0.96.8

Wyqer committed:
Update CHANGELOG.md

Wyqer committed:
Renamed objectinits, created KPLIB_whitelist file

Wyqer committed:
Increased task draw distance in description

Wyqer committed:
Changed variable name for param key and init done

Wyqer committed:
Merge branch 'master' into v0.96.8

Wyqer committed:
Variable names in new arsenal presets updated

Wyqer committed:
Some more variable adjustments from backport 7a

Wyqer committed:
Collision ignore list for building to classnameLists

Dubjunk committed:
suggestions by @Wyqer

Wyqer committed:
Fixed chernarus winter/summer ignored buildings

k4s0 committed:
[v0.96.8] update Italian localization

update Italian localization

Wyqer committed:
Merge pull request #832 from k4s0/v0.96.8

[v0.96.8] update Italian localization

Wyqer committed:
Update Changelog

veteran29 committed:
Support CBA Optics and MRT Accesories

Obsoletes #833

veteran29 committed:
Fix hardcoded class in MRT Accessory support

Nicoman55 committed:
minor bugfix

Wyqer committed:
Merge pull request #834 from nicoman35/v0.96.8

minor bugfix

Dubjunk committed:
some minor fixes

Dubjunk committed:
Merge branch 'v0.96.8' into v0.96.8-carry_ressouce_crates

Commit:
Update mission.sqm

Updated username - too many were asking me if I like My Little Pony, what even?!

Wyqer committed:
Merge pull request #844 from spawnAjak/patch-2

Update mission.sqm

LuizBarros99 commented:
> * [](https://github.com/KillahPotatoes/KP-Liberation/blob/Rewrite-icebox/Missionframework/presets/armies/customWest.sqf)
> * [ ]  Add AA/Mortar spawning to enemy bases as constant units (maybe with dynamic simulation) to create a more immersive threat in enemy territory. And to have a strategic approach to take towers and bases before attacking a city (which would cause mortar support from bases and angry civs for destroyed buildings)

Maybe instead of Dynamic Simulation, what if these units were "made virtual"? (for example what ALiVE mod does)
I could see these units being all spawned on the first sector activation of the military base, and when said base sector deactivates save "all static units are still alive". When the sector is activated once again then spawn back only the alive units once more.

I also suggest to NOT save the state of "constant units" between server restarts.

Thatguy553 committed:
Update setUnitLoadout to CBA_fnc_setLoadout

- Updated open_arsenal.sqf and redeploy_manager.sqf to use CBA_fnc_setLoadout.

- Created config file bool variable to allow easy change of CBA_fnc_setLoadout's third param to refill mags.

some-evil-kitty committed:
Replace kill manager with cba server event

some-evil-kitty committed:
fix formatting

some-evil-kitty committed:
change logic of cleanup

some-evil-kitty committed:
fix missing ; and missing param

Default corpse and wreck cleanup 20 minutes

some-evil-kitty committed:
Initial Commit

Fixed infantry battlegroups to not "lose" members, enable lambs reinforcements on them, change "reset" listener to server event

some-evil-kitty committed:
docblocked

some-evil-kitty committed:
caught more MPKilled's, killed them

also updated docs

some-evil-kitty committed:
fixed spawnvehicle, which was breaking spawn battlegroup

spawn returns a script handle, not a vehicle, meaning vehicle battlegroups lose track of their spawned vehicles

veteran29 committed:
Build tool - update package-lock.json

veteran29 committed:
Refactor build tool into a subdirectory, move build output to repo root

veteran29 committed:
Build tool - Silent npm install, npm audit fix

veteran29 committed:
Cleanup .gitignore

veteran29 committed:
VSCode - Add tasks/script for dev setup and game launch

veteran29 committed:
Fix CI

veteran29 committed:
Unify build output naming

veteran29 committed:
Add changelog entries

veteran29 committed:
Merge pull request #944 from KillahPotatoes/v0.96.8-tools

v0.96.8 - Tools improvements

veteran29 committed:
Code formatting, remove unused variables

veteran29 committed:
Revert to MPKilled, use localEvent

veteran29 committed:
Add missing params, minor formatting

veteran29 committed:
Merge remote-tracking branch 'origin/v0.96.8' into v0.96.8-carry_ressouce_crates

veteran29 committed:
Fix crates floating after detach

veteran29 committed:
Optimize crate carry/drop actions

veteran29 committed:
Prevent players from getting into vehicles while carrying

veteran29 committed:
Prevent weaponization of crates

veteran29 committed:
Merge pull request #813 from KillahPotatoes/v0.96.8-carry_ressouce_crates

[V0.96.8] carry ressource crates

veteran29 committed:
Fix vehicle unlock check error

veteran29 committed:
Add scriptName to KPLIB script for easier debugging

veteran29 committed:
Add more scriptName to KPLIB scripts for easier debugging

veteran29 committed:
Merge pull request #945 from KillahPotatoes/v0.96.8-script-names

V0.96.8 script names

some-evil-kitty committed:
Merge branch 'v0.96.8' into v0.96.8-pr-killManager

some-evil-kitty committed:
Merge pull request #939 from KillahPotatoes/v0.96.8-pr-killManager

Replace kill manager with cba server event

some-evil-kitty committed:
Merge branch 'v0.96.8' into pr/920

some-evil-kitty committed:
Docblocked second time

some-evil-kitty committed:
Conflict resolving

resolving conflictsu

some-evil-kitty committed:
Merge pull request #920 from Thatguy553/master

Update setUnitLoadout to CBA_fnc_setLoadout

some-evil-kitty committed:
Merge branch 'v0.96.8' into battlegroup-ai

some-evil-kitty committed:
Merge branch 'v0.96.8' into v0.96.8-cba_optics

some-evil-kitty committed:
Merge pull request #840 from KillahPotatoes/v0.96.8-cba_optics

V0.96.8 CBA Optics and Accessories support

veteran29 committed:
Apply suggestions from code review

veteran29 committed:
Update Missionframework/scripts/server/battlegroup/spawn_battlegroup.sqf

veteran29 committed:
Update Missionframework/scripts/server/battlegroup/spawn_battlegroup.sqf

some-evil-kitty committed:
Update Missionframework/scripts/server/battlegroup/spawn_battlegroup.sqf

Co-authored-by: Filip Maciejewski <veteran29@users.noreply.github.com>

veteran29 committed:
Fix indent

My bad

some-evil-kitty committed:
Merge branch 'v0.96.8' into battlegroup-ai

some-evil-kitty committed:
Merge pull request #942 from KillahPotatoes/battlegroup-ai

Battle Group Spawning + AI

prisonerMO committed:
Update README.md (#958)

* Yulakia (#953)

* Add files via upload

yulakia basefile

* Add files via upload

* Update README.md

added Yulakia required downloads to readme

* Update README.md

ACE update 
This includes the following compats: any CDLC, CUP, RHS, RH, RKSL, SMA (more might follow in the future).

---------

Co-authored-by: FatRefrigerator <atmartinez911@gmail.com>
Co-authored-by: some-evil-kitty <106408180+some-evil-kitty@users.noreply.github.com>

PiG13BR committed:
Update tags in 0.96.8 (#961)

* Update fn_tutorial.fsm

* Update fn_tutorial.fsm

* Update open_arsenal.sqf

some-evil-kitty committed:
fix undefined var (#962)

some-evil-kitty committed:
Fix Arsenal CBA Disposibles (#977)

some-evil-kitty committed:
Sort Loadouts Alphabetically (#979)

* sort loadouts data

* Update CHANGELOG.md

* move sort out of if statement

no need to write it down twice

TubaHorse committed:
Fixed missing comma in opfor W. German preset, line 135 (#984)
